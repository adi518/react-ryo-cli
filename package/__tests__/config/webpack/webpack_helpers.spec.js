const { cloneDeep } = require("lodash");

const SCRIPTS = require("../../../config/scripts.js");
const {
  getOutputPath,
  extendWebpack
} = require("../../../config/webpack/webpack_helpers.js");

const commonMockArgs = {
  webpackConfig: { output: {} },
  paths: {}
};

describe("extendWebpack", () => {
  test("with build (development) script", () => {
    const mockArgs = {
      ...cloneDeep(commonMockArgs),
      env: { development: true }
    };
    const result = extendWebpack(mockArgs);

    expect(result.extendedWebpackConfig === mockArgs.webpackConfig);
    expect(result.paths === mockArgs.paths);
    expect(result).toMatchSnapshot();
  });

  test("with build:production script", () => {
    const mockArgs = {
      ...cloneDeep(commonMockArgs),
      script: SCRIPTS.BUILD_PRODUCTION
    };
    const result = extendWebpack(mockArgs);

    expect(result).toMatchSnapshot();
  });

  test("getOutputPath", () => {
    expect(getOutputPath()).toMatchSnapshot();
  });
});
