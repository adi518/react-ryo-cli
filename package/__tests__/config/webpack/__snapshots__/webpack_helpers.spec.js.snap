// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extendWebpack getOutputPath 1`] = `"D:\\\\Projects\\\\react-build\\\\package\\\\out"`;

exports[`extendWebpack with build (development) script 1`] = `
Object {
  "extendeWebpackConfig": Object {
    "devtool": "eval-source-map",
    "mode": "development",
    "output": Object {
      "path": "D:\\\\Projects\\\\react-build\\\\package\\\\out",
    },
  },
  "extendedPaths": Object {
    "appBuild": "D:\\\\Projects\\\\react-build\\\\package\\\\out",
  },
}
`;

exports[`extendWebpack with build:production script 1`] = `
Object {
  "extendeWebpackConfig": Object {
    "output": Object {
      "path": "D:\\\\Projects\\\\react-build\\\\package\\\\out",
    },
  },
  "extendedPaths": Object {
    "appBuild": "D:\\\\Projects\\\\react-build\\\\package\\\\out",
  },
}
`;
